### PR TITLE
Create Earth location object using OBSGEO and MJDAVG, not MJD-OBS

### PR DESCRIPTION
**Pull request by nden**
### Description

This pull request is to address an issue with the WCS API in the case of a spectral observation.
When `OBSGEO-X/Y/Z` keywords are present the code is looking for the time of the observation
in `MJD-OBS`. Paper III defines the OBSGEO keywords this way:

```
OBSGEO-X (floating-valued)
OBSGEO-Y (floating-valued)
OBSGEO-Z (floating-valued)
are reserved to define a representative location for the observation in a standard 
terrestrial reference frame. The location values should be right-handed, geocentric, 
Cartesian coordinates valid at the epoch of MJD-AVG or DATE-AVG. 
```
MJD-AVG, according to the `Time` paper is the effective time of the observation.
This PR changes the generation of the `EarthLocation` object to use `MJD-AVG`.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by Cadair** [comment]
I wonder if it's worth falling back to mjd-obs if AVG isn't defined? The time paper is a lot newer than the OBSGEO keys?

**Comment by pllim** [comment]
Need a change log. I'll let reviewers decide if a test is needed.

File: docs/changes/wcs/12598.bugfix.rst
Comment (Cadair) [inline docs/changes/wcs/12598.bugfix.rst:1]:
```suggestion
``WCS.pixel_to_world`` now creates an ``EarthLocation`` object using ``MJD-AVG`` if present before falling back to the old behaviour of using ``MJD-OBS``.
```

File: astropy/wcs/wcsapi/fitswcs.py
Comment (Cadair) [inline astropy/wcs/wcsapi/fitswcs.py:424]: Does MJD actually use this scale argument? If so shouldn't we be pulling it from ``TIMESYS``?

(I don't really understand MJD as I have never had to use it).

**Review by Cadair — COMMENTED** [review]
A test to verify the fallback behaviour wouldn't hurt, as then it wont regress without someone noticing. I would also quite like someone else more familiar with this code (maybe @astrofrog) to have a quick look over it to make sure I am not missing something.

**Comment by astrofrog** [comment]
Is it possible to add a test for this?

File: astropy/wcs/wcsapi/fitswcs.py
Comment (nden) [reply astropy/wcs/wcsapi/fitswcs.py:424]: I don't understand it either but according to the `Time` documentation it needs a scale. Pulling it from `TIMESYS` is better. I'll add a fall back to `utc` because JWST files have the `TIMESYS` keywords in the primary header, while the rest of the WCS keywords are in the `SCI` extension header.

**Comment by nden** [comment]
I addressed all comments.

File: astropy/wcs/wcsapi/tests/test_fitswcs.py
Comment (pllim) [inline astropy/wcs/wcsapi/tests/test_fitswcs.py:592]: Why is this test commented?

Comment (nden) [reply astropy/wcs/wcsapi/tests/test_fitswcs.py]: Oops, my mistake. Thanks for catching this.

File: astropy/wcs/wcsapi/fitswcs.py
Comment (Cadair) [inline astropy/wcs/wcsapi/fitswcs.py:430]: Why does codecov think this line hasn't been covered? The test clearly should be hitting this, is there some subtle weirdness going on here? or did codecov :exploding_head:

Comment (pllim) [reply astropy/wcs/wcsapi/fitswcs.py:430]: It only counts coverage on certain jobs. So is this the test covering this line supposed to be skipped under certain conditions (e.g., remote data)?

Comment (pllim) [reply astropy/wcs/wcsapi/fitswcs.py:430]: If not, then the test is not doing what you think it is doing. I think we should make sure before merge.

Comment (nden) [reply astropy/wcs/wcsapi/fitswcs.py:430]: I added checks in the test that show that when a certain kw is `nan` the fall back is used. Regardless of what codecov says that line should be tested if i didn't miss something.

Comment (pllim) [reply astropy/wcs/wcsapi/fitswcs.py:430]: To be sure, you can run coverage locally and double check. I don't see why codecov would miss this if this test is run regardless of pytest settings.

Comment (pllim) [reply astropy/wcs/wcsapi/fitswcs.py:430]: Looks like we have two jobs that measure coverage now:

https://github.com/astropy/astropy/blob/1c5ce68e4b9569fc69d3e363a5e9f27741e95937/.github/workflows/ci_workflows.yml#L59-L62

https://github.com/astropy/astropy/blob/1c5ce68e4b9569fc69d3e363a5e9f27741e95937/.github/workflows/ci_workflows.yml#L74-L78

**Review by astrofrog — APPROVED** [review]

**Comment by Cadair** [comment]
What's the holdup here?

**Comment by pllim** [comment]
Not sure why @astrofrog only approved but did not merge.

Also there is a conflict now so this must be rebased.

**Comment by pllim** [comment]
There is a test failure.

**Comment by Cadair** [comment]
Looks like I borked the rebase.

**Comment by Cadair** [comment]
I just ran into this again with DKIST data. I fixed the bad rebase I think this is mergeable?

**Comment by nden** [comment]
Since Tom approved and it sounds like you approve, I am going to merge it.

**Pull request merged**
